src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionSymbolService.cs

```diff
@@ -12,23 +13,38 @@ internal abstract class AbstractGoToDefinitionSymbolService : IGoToDefinitionSym
     {
         protected abstract ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation);
 
-        public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, CancellationToken cancellationToken)
+        public async Task<(ISymbol, TextSpan)> GetSymbolAndBoundSpanAsync(Document document, int position, bool includeLiterals, CancellationToken cancellationToken)
         {
             var workspace = document.Project.Solution.Workspace;
 
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var semanticInfo = await SymbolFinder.GetSemanticInfoAtPositionAsync(semanticModel, position, workspace, cancellationToken).ConfigureAwait(false);
+            var symbol = GetSymbol(semanticInfo, includeLiterals);
 
-            // prefer references to declarations.  It's more likely that the user is attempting to 
-            // go to a definition at some other location, rather than the definition they're on.  
+            if (symbol is null)
+            {
+                return (null, semanticInfo.Span);
+            }
+
+            return (FindRelatedExplicitlyDeclaredSymbol(symbol, semanticModel.Compilation), semanticInfo.Span);
+        }
+
+        private ISymbol GetSymbol(TokenSemanticInfo semanticInfo, bool includeLiterals)
```

Comment:
why a new method?

Reply:
it abstracts the functionality to resolve the symbol depending on the flag